internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```diff
@@ -10,25 +14,90 @@ PublicNetworkLoadBalancer:
         - Fn::ImportValue:
             !Sub '${AppName}-${EnvName}-PublicSubnets'
     Type: network
+{{- if .NLB.UDPListenerExists }}
+    SecurityGroups:
+      - !GetAtt NetworkLoadBalancerSecurityGroup.GroupId
+NetworkLoadBalancerSecurityGroup:
+  Metadata:
+    'aws:copilot:description': 'A security group for your network load balancer allowing public traffic'
+  Type: AWS::EC2::SecurityGroup
+  Properties:
+    GroupDescription: Allow traffic to the public facing load balancer on listener ports
+    SecurityGroupIngress:
+{{- range $listener := .NLB.Listener }}
+      - CidrIp: 0.0.0.0/0
+        Description: Allow {{- if eq $listener.Protocol "TLS" }} TCP {{- else }} {{ $listener.Protocol }} {{- end}} traffic from anyone on listener port {{ $listener.Port }}
+        FromPort: {{ $listener.Port }}
+        IpProtocol: {{- if eq $listener.Protocol "TLS" }} TCP {{- else }} {{ $listener.Protocol }} {{- end}}
+        ToPort: {{ $listener.Port }}
+{{- end}}
+    VpcId:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-VpcId" 
+    Tags:
+      - Key: Name
+        Value: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-nlb'
+EnvironmentSecurityGroupIngressFromNetworkLoadBalancerSecurityGroup:
+  Type: AWS::EC2::SecurityGroupIngress
+  Metadata:
+    'aws:copilot:description': 'Allow ingress from the network load balancer to services in your environment'
+  Properties:
+    GroupId:
+      Fn::ImportValue:
+        !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
+    IpProtocol: -1
+    SourceSecurityGroupId: !GetAtt NetworkLoadBalancerSecurityGroup.GroupId
+{{- else}}
+NLBSecurityGroup:
```

Comment:
can we not drop this now? and just do the same thing we do with `EnvironmentSecurityGroupIngressFromNetworkLoadBalancerSecurityGroup`?

Reply:
I don't think so, because the only reason we can use the `EnvironmentSecurityGroupIngressFromNetworkLoadBalancerSecurityGroup` is that it accepts traffic from the newly created security group, but that one can only exist when it can be created with the NLB.

Reply:
Adding a bit more detail here!

So we are promoting to V2 for users that enable UDP, and staying in V1 for users with existing NLB and has not enabled UDP, for 2 versions. After 2 versions, the second group of users will promote to V2 as well.

Using `EnvironmentSecurityGroupIngressFromNetworkLoadBalancerSecurityGroup` is only possible for V2, i.e. group 1 users, which is exactly what the `{{if}}` statement represents. The group 2 is what the `{{else}}` statement corresponds to.